lib/charms/operator_libs_linux/v0/systemd.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Abstractions for stopping, starting and managing system services via systemd.
+
+This library assumes that your charm is running on a platform that uses systemd. E.g.,
+Centos 7 or later, Ubuntu Xenial (16.04) or later.
+
+For the most part, we transparently provide an interface to a commonly used selection of
+systemd commands, with a few shortcuts baked in. For example, service_pause and
+service_resume with run the mask/unmask and enable/disable invocations.
+
+Example usage:
+```python
+from systemd import *
+
+# Start a service
+if not service_running("mysql"):
+    success = service_start("mysql")
+
+# Attempt to reload a service, restarting if necessary
+success = service_reload("nginx", restart_on_failure=True)
+
+```
+
+"""
+
+import logging
+import subprocess
+
+__all__ = [  # Don't export `service`. (It's not the intended way of using this lib.)
+    "service_pause",
+    "service_reload",
+    "service_restart",
+    "service_resume",
+    "service_running",
+    "service_start",
+    "service_stop",
+]
+
+logger = logging.getLogger(__name__)
+
+
+def service(action: str, service_name: str) -> bool:
+    """Control a system service.
+
+    :param action: the action to take on the service
+    :param service_name: the name of the service to perform th action on
+    """
+    cmd = ["systemctl", action, service_name]
+    if action != "is-active":
+        logger.debug("Attempting to {} '{}' with command {}.".format(action, service_name, cmd))
+    else:
+        logger.debug("Checking if '{}' is active".format(service_name))
+    return subprocess.call(cmd) == 0
+
+
+def service_running(service_name: str) -> bool:
+    """Determine whether a system service is running.
+
+    :param service_name: the name of the service
+    """
+    return service("is-active", service_name)
+
+
+def service_start(service_name: str) -> bool:
+    """Start a system service.
+
+    :param service_name: the name of the service to stop
+    """
+    return service("start", service_name)
```

Comment:
Feels like each of these could do `logger.info("starting service '%s'", service_name)` or relevant message. I don't think that'd be too much and it would give some good feedback to the users of this library

Reply:
Ah yes. Silly me. Thanks